worker.py

```diff
@@ -0,0 +1,274 @@
+#!/usr/bin/env python
+
+#
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# NVIDIA CORPORATION and its licensors retain all intellectual property
+# and proprietary rights in and to this software, related documentation
+# and any modifications thereto.  Any use, reproduction, disclosure or
+# distribution of this software and related documentation without an express
+# license agreement from NVIDIA CORPORATION is strictly prohibited.
+#
+
+# system imports
+import logging
+import os
+import random
+import sys
+import tempfile
+import time
+
+import h5py
+import multiprocessing
+import numpy as np
+import pandas as pd
+import torch
+import torch.backends.cudnn as cudnn
+import torch.distributed as dist
+import torch.multiprocessing as mp
+import torch.nn as nn
+from torch.nn.parallel import DistributedDataParallel
+from torch.optim import Adam
+
+# module imports
+from claragenomics.dl4atac.dataset import DatasetTrain, DatasetInfer
+from claragenomics.dl4atac.evaluate import evaluate
+from claragenomics.dl4atac.infer import infer
+from claragenomics.dl4atac.losses import MultiLoss
+from claragenomics.dl4atac import metrics
+from claragenomics.dl4atac.metrics import BCE, MSE, Recall, Specificity, CorrCoef, AUROC
+from claragenomics.dl4atac.models.models import *
+from claragenomics.dl4atac.models.model_utils import build_model
+from claragenomics.dl4atac.train import train
+from claragenomics.dl4atac.utils import *
+from claragenomics.io.bedgraphio import expand_interval, intervals_to_bg, df_to_bedGraph
+from claragenomics.io.bigwigio import bedgraph_to_bigwig
+from cmd_args import parse_args
+
+# python imports
+import warnings
+import glob
+import math
+warnings.filterwarnings("ignore")
+
+# Set up logging
+log_formatter = logging.Formatter(
+    '%(levelname)s:%(asctime)s:%(name)s] %(message)s')
+_logger = logging.getLogger('AtacWorks-main')
+_handler = logging.StreamHandler()
+_handler.setLevel(logging.INFO)
+_handler.setFormatter(log_formatter)
+_logger.setLevel(logging.INFO)
+_logger.addHandler(_handler)
+
+
+def get_losses(task, mse_weight, pearson_weight, gpu, poisson_weight):
```

Comment:
for me the `get_metrics`, `get_losses` and `get_model` functions don't quite fit under a `workers` module, since those functions themselves are used by workers but not specific to workers. how about we leave those functions in main, and instead pass the losses/metrics/model objects into the worker functions as args?

Reply:
but they are specific to workers. They seem out of place in main.py as well. Consider these as helper functions ?

Reply:
Yeah, I think it's okay to consider them as helper functions. But since they've been moved to other modules, it would be good to make the following changes - 
1. remove the imports from main that aren't required anymore
2. update the API for the workers to take specific arguments instead of the `args` struct. since they're in a separate module now it's better to have fixed APIs for them

Reply:
keep in mind here, the args struct is the config object. So, I don't see anything wrong about passing the config object around. I removed code that modifies the args object.